server/src/@common/sentry/sentry.module.ts

```diff
@@ -0,0 +1,42 @@
+import { Logger, Module } from '@nestjs/common';
+import * as Sentry from '@sentry/node';
+import { APP_INTERCEPTOR } from '@nestjs/core';
+import { SentryPerformanceService } from './sentry-performance/sentry-performance.service';
+import { SentryInterceptor } from './sentry.interceptor';
+import { SentryExceptionService } from './sentry-exception/sentry-exception.service';
+
+export const SENTRY_OPTIONS = 'SENTRY_OPTIONS';
+
+@Module({
+    providers: [SentryPerformanceService, SentryExceptionService]
+})
+export class SentryModule {
+    static forRoot(options: Sentry.NodeOptions) {
+        if (options.dsn === 'undefined') {
+            Logger.warn(`Sentry DSN not set`, 'SentryModule');
+            options.dsn = null;
+        }
+
+        // initialization of Sentry, this is where Sentry will create a Hub
+        Logger.log(`Init sentry with these options: [${JSON.stringify(options)}]`, 'SentryModule');
+
+        Sentry.init(options);
+
+        return {
+            module: SentryModule,
+            providers: [
+                {
+                    provide: SENTRY_OPTIONS,
+                    useValue: options
+                },
+                SentryPerformanceService,
+                SentryExceptionService,
+                {
+                    provide: APP_INTERCEPTOR,
+                    useClass: SentryInterceptor
```

Comment:
Does this need to be `useClass` rather than `useValue`? Isn't SentryInterceptor a constant value? 

https://docs.nestjs.com/fundamentals/custom-providers#class-providers-useclass

Reply:
UseValue is used when you want to provide a static Object, use class is used when you want to provide a static Service or similar that has a constructor that needs to be run for the service to be useful.

https://medium.com/@paynoattn/use-usevalue-if-your-stub-is-a-object-useclass-if-your-stub-is-a-class-88ba6e1574f0
https://offering.solutions/blog/articles/2018/08/17/using-useclass-usefactory-usevalue-useexisting-with-treeshakeable-providers-in-angular/